### PR TITLE
More checks on pre-commit

### PR DESCRIPTION
**Pull request by arielsvn**
## Issue Number

N/A

## Purpose/Implementation Notes

Adds a few other pre-commit hooks for several useful things:

- block commits on dev and master (I have made them by mistake)
- Remove trailing whitespaces
- Fix end of line
- Fail if files larger than 500kb are added
- Check merge conflicts
- Check file docstrings
- check yaml syntax

These are all of the pre-commit checks that I wanted to add.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

None

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by arielsvn** [comment]
Some of these are failing, marked as WIP.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
+      - id: check-docstring-first
+      - id: check-yaml
 
   - repo: https://github.com/pre-commit/mirrors-isort
     rev: v4.3.21
     hooks:
       - id: isort
+
+  - repo: https://github.com/psf/black
+    rev: stable
```
Comment (kurtwheeler) [inline .pre-commit-config.yaml]: Can we actually pin this to a specific version so we don't have to worry about someone getting malicious code into this repo?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
-      - id: black
-        args: [--line-length=100]
+      - id: no-commit-to-branch
+        args: [--branch, dev, --branch, master]
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: [--fix=lf]
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+      - id: check-added-large-files
```
Comment (kurtwheeler) [inline .pre-commit-config.yaml]: I think we should nix this!

File: common/data_refinery_common/models/models.py [hunk 3]
```diff
@@ -1524,13 +1523,6 @@ def terms_and_conditions(self):
         return settings.TERMS_AND_CONDITIONS
 
 
-"""
-# Associations
-
-These represent the relationships between items in the other tables.
-"""
-
-
```
Comment (kurtwheeler) [inline common/data_refinery_common/models/models.py]: I don't like these changes. These docstring denoted different parts of the file.

We should probably handle this issue by breaking this file up into different namespaces, but just moving all these docstrings to the top doesn't really make much sense.

Comment (arielsvn) [reply common/data_refinery_common/models/models.py]: yeah, I wasn't sure what to do with that rule. I'll revert these and comment it until we simplify `models.py`

**Comment by arielsvn** [comment]
Addressed your comments @kurtwheeler, let me know if you see something else.

**Pull request merged**